scripts/import_data.py

```diff
@@ -0,0 +1,480 @@
+            if df[header].isnull().all():
+                headers.remove(header)
+
+    items = [{} for _ in range(df.shape[0])]
+
+    for header in headers:
+        for row in range(df.shape[0]):
+            value = df[header][row]
+            if type(value) is float:  # because pandas imports empty cells as float("nan") (even with dtype=str)
+                value = ''
+
+            if not skip_empty_fields or value.strip() != '':
+                items[row][header] = value
+
+    return items, headers, header_types
+
+
+def db_get_username(cursor: MySQLCursor) -> str:
+    """
+    Returns the username of the first Koillection user if only one user exists.
+    Otherwise, prints available users and exits.
+    :param cursor: opened cursor to the koillection database
+    :return: name of the Koillection user
+    """
+    cursor.execute("SELECT username FROM koi_user")
+    usernames = [name[0] for name in cursor.fetchall()]
+
+    if not usernames:
+        print("There should be at least one user.")
+        sys.exit(1)
+    elif len(usernames) > 1:
+        print("Argument --user is required when there are multiple Koillection users present. Available users:")
+        for name in usernames:
+            print(name)
+        sys.exit(1)
+
+    return usernames[0]
+
+
+def db_get_collection_id(cursor: MySQLCursor, collection_name: str, owner_id: str) -> str:
+    """
+    Finds a collection with the provided name. If it doesn't exist, creates a new collection belonging to owner_id.
+    Returns the collection id.
+    :param cursor: opened cursor to the koillection database
+    :param collection_name: name of the collection to find or create
+    :param owner_id: id of the collection owner (applicable if creating a new collection)
+    :return: collection id
+    """
+    cursor.execute("SELECT title FROM koi_collection")
+    collection_names = [name[0] for name in cursor.fetchall()]
+    if collection_name not in collection_names:  # create a new collection
+        collection_id = insert_collection(cursor, owner_id, collection_name)
+        insert_log(cursor, owner_id, collection_id, collection_name, "App\\Entity\\Collection")
+    else:
+        cursor.execute(f"SELECT id FROM koi_collection WHERE title='{collection_name}'")
+        collection_id = cursor.fetchone()[0]
+
+    return collection_id
+
+
+def insert_display_configuration(cursor: MySQLCursor, owner_id: str) -> str:
+    """
+    Inserts a single display configuration entry into the koi_display_configuration table and returns its id
+    :param cursor: opened cursor to the koillection database
+    :param owner_id:
+    :return: id of the created display configuration
+    """
+    display_configuration_data = {
+        "id": generate_id(),
+        "owner_id": owner_id,
+        "display_mode": "list",  # or grid
+        "sorting_direction": "ASC",
+        "created_at": get_current_time(),
+        "columns": "a:0:{}"  # I have no idea what this does
```

Comment:
Let's assume we know what is going on and remove this comment?

Please consider

Reply:
Oh, it's a remnant from a long time ago, thank you for noticing it. The column field lists all the items' columns that should be displayed in collection list view, next to the default "Name" column. (I leave it at the default a:0:{} because we don't want additional columns displayed by default)